common/source/docs/common-auxiliary-functions.rst

```diff
@@ -0,0 +1,633 @@
+.. _common-auxiliary-functions:
+
+===================
+Auxiliary Functions
+===================
+
+This feature is firmware versions 4.0 and higher. In Copter versions before 4.0 similar functions were implemented using the CHx_OPT parameters. See :ref:`Channel options <channel-7-and-8-options>` page.
+
+This page describes how to set up additional features which can be invoked from the transmitter's auxiliary function switches.
+
+Configuring which transmitter channel is used
+=============================================
+
+Any RC input channel can be assigned any auxiliary function. However, the RC channel should not be used by any other function, such as flight mode selection or flight controls. By default, channels 1-4 are used for flight controls, channel 8 is used for flight modes in Plane and Rover, as is channel 5 in Copter.
+
+Assigning the Feature
+=====================
+
+The RCx_OPTION parameters control these features. For example, the :ref:`RC7_OPTION <RC7_OPTION>` parameter controls which feature is assigned to RC input 7. Each RC channel has its
+own RCx_OPTION parameter that can be accessed in the full parameter list.
+
+
+Supported Features
+==================
+
++----------------------+----------------------------+----------+---------+---------+
+| **RCx_OPTION value** | **Feature Description**    |**Copter**|**Plane**|**Rover**|
++----------------------+----------------------------+----------+---------+---------+
+|        0             | Do Nothing (Default)       |    X     |    X    |    X    |
++----------------------+----------------------------+----------+---------+---------+
+|        2             | Flip                       |    X     |         |         |
```

Comment:
"FLIP mode" I guess..

Reply:
okay this one really confuses me....the action states that it flips and then returns to the flight mode it was in....that is not a flight mode but a temporary  action....so that is not a flight mode....but yet its listed as a flight mode on the flight mode switch, which begs the question, if I use the flight mode switch to move to this mode, what does it return to? the mode it was in before the switch occurred? what happens if I am a bit slow and the switch moves thru AUTO from    ACRO before getting to FLIP, does it return to AUTO? intuitively to me,  flip or brake should be actions modifying the current  flight modes, not a flight mode itself....but I need to understand the answers to the above question before I agree its a flight mode

Reply:
When flip starts it [records the mode that it was entered from](https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/mode_flip.cpp#L61) and then returns to that after it completes the flip.  So, yes, if the pilot was slow on the switch and the vehicle entered Auto before flip then it would return to Auto.

"temporary action" or "flight mode", it's implemented as a flight mode... and it can be activated from a flight mode switch or an auxiliary switch..

Reply:
fixed